engine/src/main/java/org/terasology/world/chunks/internal/ChunkImpl.java

```diff
@@ -64,6 +65,9 @@
     private static final DecimalFormat PERCENT_FORMAT = new DecimalFormat("0.##");
     private static final DecimalFormat SIZE_FORMAT = new DecimalFormat("#,###");
 
+    private final int verticalChunkMeshSegments = CoreRegistry.get(Config.class).getSystem().getVerticalChunkMeshSegments();
+    private final int chunkHalfHeight = ChunkConstants.SIZE_Y / verticalChunkMeshSegments / 2;
```

Comment:
That isn't quite correct, because we don't join the sides of adjecent blocks (due to texture rendering concerns). The total number of vertices and triangles will be the same regardless of whether there is one mesh or more. The advantage or larger mesh is less draw calls, the advantage of smaller mesh is they are quicker to update and more of the scene can be culled - although the quicker to update doesn't apply here.

VERTICAL_SEGMENTS is a really old debug setting, presumable from begla testing to see what performs best.